baremetal/metal3machine_manager.go

```diff
@@ -861,6 +864,10 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmov1alpha1.BareMetal
 			} else if !m.nodeReuseLabelExists(ctx, &host) {
 				switch host.Status.Provisioning.State {
```

Comment:
We have a case now where all options are continue, which is same as removing the switch and just executing continue directly. Even the default option is continue.

Reply:
This is actually wrong, in Go the empty case is not falling thru. We added a comment to make this more explicit.